tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12296,6 +12317,7 @@ void hlsl::HandleDeclAttributeForHLSL(Sema &S, Decl *D, const AttributeList &A,
 
   if (declAttr != nullptr)
   {
+    S.DiagnoseHLSLDeclAttr(D, declAttr);
```

Comment:
I expect this is called here instead of in `DiagnoseHLSLDecl` because the needed information is available here and not there. Nevertheless, the name of that function seems more consistent with what this does.

Reply:
There is similarity in the names but the do different things. `DiagnoseHLSLDecl` diagnoses issues with the Decl itself, whereas this is diagnosing the Decl's attributes. To diagnose the Decl's attributes we need the attribute. The new method gets called once from the Handle path during parsing the attributes and once during the instantiation path for template-dependent attributes.

Do you have a suggestion for a clearer way to name the new method to make it more distinct?

Reply:
Nah. I'm satisfied with your explanation.